### ISSUE TITLE
Question: how to debug locals evaluations

### ISSUE THREAD

**Issue by tgourley01** [gruntwork-io/terragrunt#1243]
When developing terragrunt code it can be very challenging to know how locals are being evaluated.

Even with TG_LOG=DEBUG, you can see that locals are being evaluated, but not what values they resolve to.

I've had to resort to passing local values to read_terragrunt_config() so when it fails I can at least see the value printed in the outputs.

Is there something obvious I'm missing here?

This is worse than println() debugging because there is no println() function!

**Comment by yorinasub17** [comment]
Yup this is something we want to address, but we have been super busy with other initiatives internally that has made it difficult for us to pick this up. You can see our current thinking and design on improved debugging support in https://github.com/gruntwork-io/terragrunt/pull/1137.

**Comment by dmattia** [comment]
Here's my attempt at a `print` equivalent function that works for me:

```hcl
locals {
  raw_json = "{\"foo\": \"bar\"}"
  parsed = jsondecode(local.raw_json)
  this_is_valid = local.parsed.foo
  debug = run_cmd("echo", "${jsonencode(local.parsed)}")
  this_should_error = local.parsed.does_not_exist
}
```

By adding in a `run_cmd("echo", ...)` in a locals block somewhere, you can see the value in the terminal at runtime

<img width="606" alt="Screen Shot 2021-01-28 at 2 47 51 AM" src="https://user-images.githubusercontent.com/8922077/106112731-3b2cd980-6113-11eb-84c0-face60bcd861.png">
